core/src/main/java/org/elasticsearch/tasks/TaskId.java

```diff
@@ -59,9 +70,30 @@ public TaskId(String taskId) {
         }
     }
 
-    public TaskId(StreamInput in) throws IOException {
-        nodeId = in.readString();
-        id = in.readLong();
+    /**
+     * Read a {@linkplain TaskId} from a stream. {@linkplain TaskId} has this rather than the usual constructor that takes a
+     * {@linkplain StreamInput} so we can return the {@link #EMPTY_TASK_ID} without allocating.
+     */
+    public static TaskId readFromStream(StreamInput in) throws IOException {
+        String nodeId = in.readString();
+        if ("".equals(nodeId)) {
```

Comment:
Would it be more consistent and perhaps faster to use `nodeId.isEmpty()` checks everywhere instead of `nodeId.length() == 0` or `"".equals(nodeId)`?